arch/arm/src/common/up_pthread_start.c

```diff
@@ -80,6 +80,8 @@ void up_pthread_start(pthread_startroutine_t entrypt, pthread_addr_t arg)
   /* Let sys_call2() do all of the work */
 
   sys_call2(SYS_pthread_start, (uintptr_t)entrypt, (uintptr_t)arg);
+
+  PANIC();
```

Comment:
But don't see who declare up_pthread_start/up_task_start as noreturn_function

Reply:
include/nuttx/arch.h has their prototypes with noreturn_function.

Reply:
If so, we need to change all location to ensure the warning get clear for all arch

Reply:
done